src/Symfony/Component/Console/Command/Command.php

```diff
@@ -87,10 +87,31 @@ public static function getDefaultDescription(): ?string
      *
      * @throws LogicException When the command name is empty
      */
-    public function __construct(?string $name = null)
+    public function __construct(?string $name = null, ?callable $code = null)
     {
         $this->definition = new InputDefinition();
 
+        if ($name === null && $code !== null) {
+            if (!\is_object($code) || $code instanceof \Closure) {
+                throw new InvalidArgumentException(\sprintf('The command must be an instance of "%s" or an invokable object.', Command::class));
+            }
+
+            /** @var AsCommand $attribute */
+            $attribute = ((new \ReflectionObject($code))->getAttributes(AsCommand::class)[0] ?? null)?->newInstance()
+                ?? throw new LogicException(\sprintf('The command must use the "%s" attribute.', AsCommand::class));
+
+            $this->setName($attribute->name)
+                ->setDescription($attribute->description ?? '')
+                ->setHelp($attribute->help ?? '')
+                ->setCode($code);
+
+            foreach ($attribute->usages as $usage) {
+                $this->addUsage($usage);
+            }
+
+            return;
+        }
```

Comment:
Like this, the `$name` can be overridden in the constructor.

```suggestion
        if ($code !== null) {
            if (!\is_object($code) || $code instanceof \Closure) {
                throw new InvalidArgumentException(\sprintf('The command must be an instance of "%s" or an invokable object.', Command::class));
            }

            /** @var AsCommand $attribute */
            $attribute = ((new \ReflectionObject($code))->getAttributes(AsCommand::class)[0] ?? null)?->newInstance()
                ?? throw new LogicException(\sprintf('The command must use the "%s" attribute.', AsCommand::class));

            $this->setName($name ?? $attribute->name)
                ->setDescription($attribute->description ?? '')
                ->setHelp($attribute->help ?? '')
                ->setCode($code);

            foreach ($attribute->usages as $usage) {
                $this->addUsage($usage);
            }

            return;
        }
```

Reply:
Done